integreat_compass/cms/views/offers/offer_report_view.py

```diff
@@ -0,0 +1,87 @@
+from django.contrib import messages
+from django.shortcuts import redirect, render
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...forms import ReportForm
+from ...models import Offer, OfferVersion
+
+
+class OfferReportView(TemplateView):
+    """
+    View to report an offer
+    """
+
+    template_name = "offers/offer_report.html"
+
+    @staticmethod
+    def _get_form():
+        """
+        This function gets the form Report
+
+        :return: the report form
+        :rtype: dict
+        """
+        form = ReportForm()
+        return form
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        This functions displays all offers
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: The context of all
+        :rtype: dict
+        """
+        offer = Offer.objects.get(id=kwargs.get("pk"))
+        context = {"selected_offer": offer, "form": self._get_form()}
+        return render(request, self.template_name, context)
```

Comment:
```suggestion
    def get_context_data(self, **kwargs):
        r"""
        Add context data for the offer report view

        :param \**kwargs: The supplied keyword arguments
        :type \**kwargs: dict

        :return: The context used in the offer report view
        :rtype: dict
        """
        return super().get_context_data(**kwargs) | {
            "selected_offer": Offer.objects.get(id=kwargs.get("pk")),
            "form": ReportForm(),
        }
```
`get_context_data` should be enough

Reply:
I just realized you also need to check if the PK exists, currently a DoesNotExist exception is thrown. `.filter(id=kwargs.get("pk")).first()` could be a drop-in replacement for `.get(...)`, I don't know if we have to handle users artificially changing the PK in their address bar beyond that...

Reply:
I changed it to get_object_or_404. I hope that is sufficient